File: python_leaoai/python_leaoai/services/ApplicationServices/Content/PagesService.py
from typing import List, Optional, Type, TypeVar
from sqlalchemy.orm import Session
from math import ceil
from models import Pagina, PaginaContent, Product, Servico, Artigo, Produto, Foto, Curso, Categoria  # Seus modelos SQLAlchemy
from dtos import PageDTO, ListPageDTO, Item, Pagination
from repositories import Repository  # Repositórios genéricos
from services import ItemService, PageSectionService, PageSectionItemService, ImagemFactory

T = TypeVar('T')

class PagesService:
    def __init__(
        self,
        db: Session,
        item_service: ItemService,
        pagina_repo: Repository,
        pagina_content_repo: Repository,
        banner_repo: Repository,
        categoria_repo: Repository,
        equipe_repo: Repository,
        cliente_repo: Repository,
        marca_repo: Repository,
        depoimento_repo: Repository,
        artigo_repo: Repository,
        blog_repo: Repository,
        product_repo: Repository,
        imagem_repo: ImagemFactory,
        page_section_service: PageSectionService,
        page_section_item_service: PageSectionItemService,
        servico_repo: Repository,
        produto_repo: Repository,
        foto_repo: Repository,
        curso_repo: Repository
    ):
        self._db = db
        self._srvItem = item_service
        self._pagina_repo = pagina_repo
        self._pagina_content_repo = pagina_content_repo
        self._imagem_repo = imagem_repo
        self._page_section_service = page_section_service
        self._page_section_item_service = page_section_item_service
        self._product_repo = product_repo
        self._servico_repo = servico_repo
        self._artigo_repo = artigo_repo
        self._produto_repo = produto_repo
        self._foto_repo = foto_repo
        self._curso_repo = curso_repo
        self._categoria_repo = categoria_repo

        # Map de content builders
        self._content_builders = {
            "Artigo": lambda chave, pagina_id, selector, ignore_fields: self.build_content(artigo_repo, chave, pagina_id, selector, ignore_fields),
            "Servico": lambda chave, pagina_id, selector, ignore_fields: self.build_content(servico_repo, chave, pagina_id, selector, ignore_fields),
            "Banner": lambda chave, pagina_id, selector, ignore_fields: self.build_content(banner_repo, chave, pagina_id, selector, ignore_fields),
            "Blog": lambda chave, pagina_id, selector, ignore_fields: self.build_content(blog_repo, chave, pagina_id, selector, ignore_fields),
            "Depoimento": lambda chave, pagina_id, selector, ignore_fields: self.build_content(depoimento_repo, chave, pagina_id, selector, ignore_fields),
            "Equipe": lambda chave, pagina_id, selector, ignore_fields: self.build_content(equipe_repo, chave, pagina_id, selector, ignore_fields),
            "Marca": lambda chave, pagina_id, selector, ignore_fields: self.build_content(marca_repo, chave, pagina_id, selector, ignore_fields),
            "Cliente": lambda chave, pagina_id, selector, ignore_fields: self.build_content(cliente_repo, chave, pagina_id, selector, ignore_fields),
            "Product": lambda chave, pagina_id, selector, ignore_fields: self.build_content(product_repo, chave, pagina_id, selector, ignore_fields),
            "Produto": lambda chave, pagina_id, selector, ignore_fields: self.build_content(produto_repo, chave, pagina_id, selector, ignore_fields),
            "Foto": lambda chave, pagina_id, selector, ignore_fields: self.build_content(foto_repo, chave, pagina_id, selector, ignore_fields),
            "Curso": lambda chave, pagina_id, selector, ignore_fields: self.build_content(curso_repo, chave, pagina_id, selector, ignore_fields)
        }

    def build_page(self, url: str) -> PageDTO:
        dto = PageDTO()
        pagina = self._pagina_repo.get_by_url(url)
        if not pagina:
            raise Exception("Page not found")

        dto.page = self._srvItem.build(pagina)

        contents = [c for c in self._pagina_content_repo.get_all() if c.pagina_id == pagina.id]
        for content in contents:
            builder = self._content_builders.get(content.table_action)
            if builder:
                built_content = builder(content.chave, content.pagina_id, content.selector, content.ignore_fields)
                setattr(dto, content.table_action.lower() + 's', built_content)

        dto.sections = self.get_page_sections(pagina.id, "Pagina")
        return dto

    def get_group(self, repo: Repository, category: Optional[str], search: Optional[str], tags: Optional[str],
                  page: Optional[int], destaque: Optional[bool], destaque_vitrine: Optional[bool], page_size: int = 12) -> ListPageDTO:
        entities = repo.get_all_ativo() if category in (None, "GetAll") else repo.get_all_by_categoria_url(category)

        if search:
            search_lower = search.lower()
            entities = [e for e in entities if search_lower in (getattr(e, "titulo", "") or "").lower() or
                        search_lower in (getattr(e, "descricao", "") or "").lower()]

        if destaque:
            entities = [e for e in entities if getattr(e, "destaque", False)]
        if destaque_vitrine:
            entities = [e for e in entities if getattr(e, "destaque_vitrine", False)]

        # Tags
        all_tags = set(tag for e in entities if getattr(e, "descricao", None) for tag in getattr(e, "descricao", "").split(';'))
        dto = ListPageDTO()
        dto.page.page_title = "Group"
        dto.page.tags = ";".join(all_tags)

        # Categorias
        categorias = {getattr(e, "categoria") for e in entities if getattr(e, "categoria", None)}
        dto.categorias = [Item(titulo=c.titulo, url=c.url) for c in categorias]

        # Ordenação
        entities = sorted(entities, key=lambda e: getattr(e, "id", 0), reverse=True)

        # Pagination
        page_number = page or 1
        total_items = len(entities)
        start = (page_number - 1) * page_size
        end = start + page_size
        dto.pages = [self._srvItem.build(e) for e in entities[start:end]]

        total_pages = ceil(total_items / page_size)
        dto.pagination = Pagination(
            page_size=page_size,
            page_total=total_items,
            page_number=page_number,
            is_last_page=page_number >= total_pages,
            is_first_page=page_number == 1,
            has_previous_page=page_number > 1,
            has_next_page=page_number < total_pages,
            start_page=1,
            end_page=total_pages,
            total_pages=total_pages,
            total_items=len(dto.pages),
            pages=list(range(1, total_pages + 1))
        )
        return dto

    def build_content(self, repo: Repository, chave: str, pagina_id: int, selector: str, ignore_fields: Optional[str] = None) -> List[Item]:
        items = []
        entities = repo.get_all_ativo()
        for e in entities:
            add = False
            if selector == "GetAll":
                add = True
            elif selector == "GetAllWithKey" and chave:
                add = getattr(e, "chave", None) == chave
            elif selector == "GetByPaginaId" and getattr(e, "pagina_id", None) == pagina_id:
                add = True
            elif selector == "GetByPaginaIdAndChave" and chave and getattr(e, "pagina_id", None) == pagina_id:
                add = getattr(e, "chave", None) == chave

            if add:
                obj = self._srvItem.build(e, ignore_fields.split(';') if ignore_fields else None)
                # Imagens
                imagens = self._imagem_repo.get_imagens_by_table_action_and_table_id(type(e).__name__ + "s", obj.id)
                obj.imagens = [self._srvItem.build(img) for img in imagens]
                items.append(obj)
        return items

    def get_page_sections(self, id_: int, owner_type: str) -> List[Item]:
        sections = self._page_section_service.get_page_sections_by_owner(id_, owner_type)
        section_items = []
        for s in sections:
            section_item = Item(
                ref=s.ref,
                titulo=s.titulo,
                subtitulo=s.subtitulo,
                descricao=s.descricao,
                imagem=s.imagem,
                arquivo=s.arquivo,
                thumbnail=s.thumbnail
            )
            for i in self._page_section_item_service.get_page_section_items(s.id):
                item_obj = Item(
                    ref=i.ref,
                    titulo=i.titulo,
                    subtitulo=i.subtitulo,
                    descricao=i.descricao,
                    imagem=i.imagem,
                    arquivo=i.arquivo,
                    thumbnail=s.thumbnail
                )
                section_item.items.append(item_obj)
            section_items.append(section_item)
        return section_items

File: Framework.Services/ApplicationServices/Content/PagesService.cs
using Data.Models.Conteudo;
using Data;
using Framework.Repositories;
using Framework.Data.Models.Core;
using Framework.Domain.Site.Core;
using System.Linq;
using System.Collections.Generic;
using Framework.Domain.Site.Component;
using Data.Models.Catalogo;
using Data.Models.Pessoa;
using Framework.Domain.Core;
using Framework.Factories.Core;
using Framework.Services.ApplicationServices.Core;
using Framework.Services.Interfaces.Core;
using Data.Models.Ecommerce;
using PagedList.Core;
using Data.Models.Propriedade;
using System.Data.Entity.Core;

namespace Framework.Services.ApplicationServices.Content
{
    public class PagesService : GenericService<Pagina, IRepository<Pagina>>
    {
        private readonly ApplicationDbContext _context;
        private readonly ItemService _srvItem;
        private readonly IRepository<Pagina> _repository;
        private readonly IRepository<PaginaContent> _paginaContentRepository;
        private readonly ImagemFactory _imagemRepository;
        private readonly Dictionary<string, Func<string, int, string, string, List<Item>>> _contentBuilders;
        private readonly IPageSectionService _pageSectionService;
        private readonly IPageSectionItemService _pageSectionItemService;
        private readonly IRepository<Product> _productRepository;
        private readonly IRepository<Servico> _servicoRepository;
        private readonly IRepository<Artigo> _artigoRepository;
        private readonly IRepository<Produto> _produtoRepository;
        private readonly IRepository<Foto> _fotoRepository;
        private readonly IRepository<Curso> _cursoRepository;
		private readonly IRepository<Categoria> _categoriaRepository;
        public PagesService(ApplicationDbContext context,
            ItemService itemService,
            IRepository<Pagina> repository,
            IRepository<PaginaContent> paginaContentRepository,
            IRepository<Banner> bannerRepository,
			  IRepository<Categoria> categoriaRepository,
			IRepository<Equipe> equipeRepository,
            IRepository<Cliente> clienteRepository,
            IRepository<Marca> marcaRepository,
            IRepository<Depoimento> depoimentoRepository,
            IRepository<Artigo> artigoRepository,
            IRepository<Blog> blogRepository,
            IRepository<Product> productRepository,
            ImagemFactory imagemRepository,
            IPageSectionService pageSectionService,
            IPageSectionItemService pageSectionItemService,
            IRepository<Servico> servicoRepository,
            IRepository<Produto> produtoRepository,
            IRepository<Foto> fotoRepository,
            IRepository<Curso> cursoRepository
            )
            : base(context, itemService, repository)
        {
            _context = context;
            _srvItem = itemService;
            _repository = repository;
            _paginaContentRepository = paginaContentRepository;
            _imagemRepository = imagemRepository;
            _pageSectionService = pageSectionService;
            _pageSectionItemService = pageSectionItemService;
            _productRepository = productRepository;
            _servicoRepository = servicoRepository;
            _artigoRepository = artigoRepository;
            _productRepository = productRepository;
            _produtoRepository = produtoRepository;
            _fotoRepository = fotoRepository;
            _cursoRepository = cursoRepository;
			_categoriaRepository = categoriaRepository;

			_contentBuilders = new Dictionary<string, Func<string, int, string, string, List<Item>>>
            {
                { "Artigo", (chave, paginaId, selector,ignoreFields) => BuildContent(artigoRepository, chave, paginaId, selector, ignoreFields) },
				{ "Servico", (chave, paginaId, selector,ignoreFields) => BuildContent(servicoRepository, chave, paginaId, selector, ignoreFields) },
				{ "Banner", (chave, paginaId, selector, ignoreFields) => BuildContent(bannerRepository, chave, paginaId, selector, ignoreFields) },
                { "Blog", (chave, paginaId, selector, ignoreFields) => BuildContent(blogRepository, chave, paginaId, selector, ignoreFields) },
                { "Depoimento", (chave, paginaId, selector, ignoreFields) => BuildContent(depoimentoRepository, chave, paginaId, selector, ignoreFields) },
                { "Equipe", (chave, paginaId, selector, ignoreFields) => BuildContent(equipeRepository, chave, paginaId, selector, ignoreFields) },
                { "Marca", (chave, paginaId, selector, ignoreFields) => BuildContent(marcaRepository, chave, paginaId, selector, ignoreFields) },
                { "Cliente", (chave, paginaId, selector, ignoreFields) => BuildContent(clienteRepository, chave, paginaId, selector, ignoreFields) },
                { "Product", (chave, paginaId, selector, ignoreFields) => BuildContent(productRepository, chave, paginaId, selector, ignoreFields) },
                { "Produto", (chave, paginaId, selector, ignoreFields) => BuildContent(produtoRepository, chave, paginaId, selector, ignoreFields) },
                { "Foto", (chave, paginaId, selector, ignoreFields) => BuildContent(fotoRepository, chave, paginaId, selector, ignoreFields) },
                { "Curso", (chave, paginaId, selector, ignoreFields) => BuildContent(cursoRepository, chave, paginaId, selector, ignoreFields) }
            };
        }
        public PageDTO BuildPage(string url)
        {
            PageDTO dto = new PageDTO();

            try
            {
                Pagina pagina = _repository.GetByUrl(url);

                if (pagina is null)
                    throw new EntityException("Page not found");

                dto.Page = _srvItem.Build(pagina);

                var contents = _paginaContentRepository.GetAll()
                    .Where(b => b.PaginaId == pagina.Id)
                    .ToList();

                foreach (var item in contents)
                {
                    if (_contentBuilders.TryGetValue(item.TableAction, out var builder))
                    {
                        // Passe um valor adequado para o selector conforme sua lógica
                        var content = builder(item.Chave, item.PaginaId, item.Selector, item.IgnoreFields);

                        switch (item.TableAction)
                        {
                            case "Artigo":
                                dto.Artigos = content;
                                break;
                            case "Banner":
                                dto.Banners = content;
                                break;
                            case "Blog":
                                dto.Posts = content;
                                break;
                            case "Depoimento":
                                dto.Depoimentos = content;
                                break;
                            case "Equipe":
                                dto.Equipes = content;
                                break;
                            case "Marca":
                                dto.Marcas = content;
                                break;
                            case "Cliente":
                                dto.Clientes = content;
                                break;
                            case "Produto":
                                dto.Produtos = content;
                                break;
							case "Servico":
								dto.Servicos = content;
								break;
							case "Product":
                                dto.Products = content;
                                break;
                            case "Foto":
                                dto.Fotos = content;
                                break;
                            case "Curso":
                                dto.Cursos = content;
                                break;
                        }
                    }
                }

                dto.Sections = GetPageSections(pagina.Id, "Pagina");


            }
            catch (Exception ex)
            {

            }

            return dto;

        }


        public ListPageDTO GetGroup<T>(IRepository<T> repository, string category, string search, string tags, int? page, bool? destaque, bool? destaqueVitrine, int pageSize = 12) where T : class
        {
            List<Categoria> categories = _categoriaRepository.GetAllAtivo().ToList();

            if (!search.IsNullOrEmpty())
                search = search.ToLower();

            #region Initialize
            ListPageDTO dto = new ListPageDTO();
            int Pagesize = pageSize;
            int PageNumber = page == 0 ? 1 : page ?? 1;
			#endregion

			#region Filter Category
			IEnumerable<T> entities;

			switch (category)
			{
				case null:
				case "GetAll":
					entities = repository.GetAllAtivo();
					break;

				default:
					entities = repository.GetAllByCategoriaUrl(category);
					break;
			}
			#endregion

			#region Search
			if (!string.IsNullOrEmpty(search))
                entities = entities.Where(e => typeof(T).GetProperty("Titulo")?.GetValue(e)?.ToString()?.ToLower().Contains(search) == true ||
                                               typeof(T).GetProperty("Descricao")?.GetValue(e)?.ToString()?.ToLower().Contains(search) == true);
            #endregion

            #region Destaque e DestaqueVitrine
            if (destaque == true)
            {
                var destaqueProperty = typeof(T).GetProperty("Destaque");

                if (destaqueProperty != null)
                {
                    entities = entities.Where(e =>
                        destaqueProperty.GetValue(e) is bool value && value);
                }
            }
            if (destaqueVitrine == true)
            {
                var destaqueVitrineProperty = typeof(T).GetProperty("DestaqueVitrine");

                if (destaqueVitrineProperty != null)
                {
                    entities = entities.Where(e =>
                        destaqueVitrineProperty.GetValue(e) is bool value && value);
                }
            }
            #endregion

            var allTags = entities
                 .Where(entity => typeof(T).GetProperty("Descricao")?.GetValue(entity) != null)
                 .SelectMany(entity => typeof(T).GetProperty("Descricao")?.GetValue(entity)?.ToString().Split(';') ?? Array.Empty<string>())
                 .Distinct()
                 .ToArray();

            dto.Page.PageTitle = "Group";
            dto.Page.Tags = string.Join(";", allTags);

            var categorias = entities
                                     .Select(e => typeof(T).GetProperty("Categoria")?.GetValue(e))
                                     .Where(c => c != null)
                                     .Distinct()
                                     .ToList();

            foreach (Categoria item in categorias)
            {
                dto.Categorias.Add(new Item()
                {
                    Titulo = item.Titulo,
                    Url = item.Url,
                });
            }

            entities = entities.OrderByDescending(e => (int?)typeof(T).GetProperty("Id")?.GetValue(e)).ToList();

            #region Creator
            IQueryable<Item> EntityPosts = Creator(entities.ToList());
            #endregion

            #region Pagination
            dto.Pages = EntityPosts.ToPagedList(PageNumber, Pagesize);

            Pagination pagination = new Pagination
            {
                PageSize = Pagesize,
                PageTotal = entities.Count(),
                PageNumber = dto.Pages.PageNumber,
                IsLastPage = dto.Pages.IsLastPage,
                IsFirstPage = dto.Pages.IsFirstPage,
                HasPreviousPage = dto.Pages.HasPreviousPage,
                HasNextPage = dto.Pages.HasNextPage,
                StartPage = 1,
                totalPages = dto.Pages.PageCount,
                totalItems = dto.Pages.Count,
                EndPage = dto.Pages.PageCount,
                pages = Enumerable.Range(1, dto.Pages.PageCount).ToArray()
            };

            dto.Pagination = pagination;
            #endregion

            return dto;
        }

        private IQueryable<Item> Creator<T>(List<T> entities) where T : class
        {
            List<Item> list = new List<Item>();

            foreach (var entity in entities)
            {
                list.Add(_srvItem.Build(entity));
            }

            return list.AsQueryable();
        }

        public ListPageDTO GetItems(string entityKey, string category, string search, string tags, int? page, bool? destaque, bool? destaqueVitrine, int pageSize = 32)
        {
            switch (entityKey)
            {
				case "categories":
					return GetGroup(_categoriaRepository, category, search, tags, page, destaque, destaqueVitrine, pageSize);
				case "products":
                    return GetGroup(_productRepository, category, search, tags, page, destaque, destaqueVitrine, pageSize);
                case "produtos":
                    return GetGroup(_produtoRepository, category, search, tags, page, destaque, destaqueVitrine, pageSize);
                case "services":
                    return GetGroup(_servicoRepository, category, search, tags, page, destaque, destaqueVitrine);
                case "artigos":
                    return GetGroup(_servicoRepository, category, search, tags, page, destaque, destaqueVitrine);
                default:
                    return new ListPageDTO();
            }
        }


        #region Groups

        public List<Item> GetPageSections(int Id, string OwnerType)
        {
            // Fetch PageSections
            var pageSections = _pageSectionService.GetPageSectionsByOwner(Id, OwnerType);

            var pageSectionItems = new List<Item>();

            foreach (var section in pageSections)
            {
                var sectionItem = new Item
                {
                    Ref = section.Ref,
                    Titulo = section.Titulo,
                    Subtitulo = section.Subtitulo,
                    Descricao = section.Descricao,
                    Imagem = section.Imagem,
                    Arquivo = section.Arquivo,
                    Thumbnail = section.Thumbnail
                };

                // Fetch PageSectionItems
                var items = _pageSectionItemService.GetPageSectionItems(section.Id);

                foreach (var item in items)
                {
                    var itemObj = new Item
                    {
                        Ref = item.Ref,
                        Imagem = item.Imagem,
                        Titulo = item.Titulo,
                        Subtitulo = item.Subtitulo,
                        Descricao = item.Descricao,
                        Arquivo = item.Arquivo,
                        Thumbnail = section.Thumbnail
                    };

                    sectionItem.Items.Add(itemObj);
                }

                pageSectionItems.Add(sectionItem);
            }

            return pageSectionItems;
        }

     
        //public ListPageDTO GetGroupPage(string url)
        //{
        //    ListPageDTO dto = new ListPageDTO();

        //    foreach (var item in _repository.GetAllAtivo().Where(b => b.GroupPagina == url).ToList())
        //    {
        //        dto.Pages.Add(new PageItemDTO()
        //        {
        //            Id = item.Id,
        //            Link = item.Url,
        //            Ordem = item.Ordem,
        //            Titulo = item.Titulo
        //        });
        //    };

        //    return dto;
        //}

        private List<Item> BuildContent<T>(IRepository<T> repository, string chave, int paginaId, string selector, string ignoreFields = "") where T : class
        {
            var dto = new List<Item>();
            var items = repository.GetAllAtivo().ToList();

            foreach (var item in items)
            {
                var itemType = item.GetType();
                bool isAbleToAdd = false;

                if (!string.IsNullOrEmpty(selector) && selector == "GetAll")
                {
                    isAbleToAdd = true;
                }
                else if (!string.IsNullOrEmpty(selector) && selector == "GetAllWithKey" && !string.IsNullOrEmpty(chave))
                {
                    var chaveProperty = itemType.GetProperty("Chave");
                    if (chaveProperty != null)
                    {
                        var itemChave = chaveProperty?.GetValue(item)?.ToString();

                        if (itemChave == chave)
                            isAbleToAdd = true;
                    }
                }
                else if (!string.IsNullOrEmpty(selector) && selector == "GetByPaginaId" && paginaId > 0)
                {
                    var paginaIdProperty = itemType.GetProperty("PaginaId");

                    if (paginaIdProperty != null)
                    {
                        var itemPaginaId = (int?)paginaIdProperty?.GetValue(item);

                        if (itemPaginaId > 0 && itemPaginaId == paginaId)
                            isAbleToAdd = true;
                    }
                }
                else if (!string.IsNullOrEmpty(selector) && selector == "GetByPaginaIdAndChave" && !string.IsNullOrEmpty(chave) && paginaId > 0)
                {
                    var paginaIdProperty = itemType.GetProperty("PaginaId");

                    if (paginaIdProperty != null)
                    {
                        var itemPaginaId = (int?)paginaIdProperty?.GetValue(item);

                        if (itemPaginaId > 0 && itemPaginaId == paginaId)
                        {
                            var chaveProperty = itemType.GetProperty("Chave");
                            if (chaveProperty != null)
                            {
                                var itemChave = chaveProperty?.GetValue(item)?.ToString();

                                if (itemChave == chave)
                                    isAbleToAdd = true;
                            }
                        }
                    }
                }

                if (isAbleToAdd)
                {
                    Item obj = _srvItem.Build(item, ignoreFields.IsNullOrEmpty() ? null : ignoreFields.Split(';').ToList());

                    Type itemTypeT = typeof(T);

                    try
                    {
                        var Imagens = _imagemRepository.GetImagemsByTableActionAndTableId(itemTypeT.Name.ToString() + "s", obj.Id.Value);

                        foreach (var imagem in Imagens)
                        {
                            obj.imagens.Add(_srvItem.Build(imagem));
                        }
                    }
                    catch (Exception ex)
                    {
                    }


                    dto.Add(obj);

                }
            }

            return dto;
        }

        public PageDTO GetDetailGroup(string entityKey, string url)
        {
            PageDTO dto = new PageDTO();

            switch (entityKey)
            {
                case "cursos":
                    Curso curso = _cursoRepository.GetByUrl(url);
                    dto.Page = _srvItem.Build(curso);
                    dto.Sections = GetPageSections(curso.Id, "Cursos");
                    break;
                case "products":
                    Product product = _productRepository.GetByUrl(url);
                    dto.Page = _srvItem.Build(product);
                    dto.Sections = GetPageSections(product.Id, "Products");
                    break;

                case "produtos":
                    Produto produto = _produtoRepository.GetByUrl(url);
                    dto.Page = _srvItem.Build(produto);
                    dto.Sections = GetPageSections(produto.Id, "Produtos");
                    break;
                case "services":
					Servico servico = _servicoRepository.GetByUrl(url);
					dto.Page = _srvItem.Build(servico);
					dto.Sections.AddRange(GetPageSections(servico.Id, "Servicos"));
					dto.Sections.AddRange(GetPageSections(servico.Id, "Servico"));
					break;
                case "artigos": dto.Page = _srvItem.Build(_artigoRepository.GetByUrl(url)); break;

                default:
                    break;
            }

            return dto;
        }

        #endregion
  
    }
}
